src/TConsole.cpp

```diff
@@ -1934,35 +1935,39 @@ bool TConsole::moveCursor(int x, int y)
     }
 }
 
-int TConsole::select(const QString& text, int numOfMatch)
+// Previously -1 was used as a general error including the string not being
+// found - now it is only used for that case, any other returns -2 and an
+// error message for the lua caller:
```

Comment:
Consulting the API in the wiki suggests I am going to have to rethink this:
>  To prevent working on random text if your selection didn't actually select anything, check the -1 return code before doing changes

So - perhaps I should stick to the `-1` return in all cases (though I could still return an extra error message as a second return value...?

Reply:
Yes, -1 + msg sounds good

Reply:
Done

Reply:
Do we really need a comment about the history of the code here?